### PR TITLE
Removed deprecated docker email flag

### PR DESCRIPTION
**Pull request by xavi-**
It's now breaking builds: https://circleci.com/gh/Clever/kinesis-to-firehose/136

cc @nathanleiby

### CODE REVIEW

**Comment by nathanleiby** [comment]
If it's only breaking the build on that one repo, we should test before rolling it out more widely. That repo is using CircleCI 2.

A possible way to target just CircleCI2 would be to copy-paste this script into a new base folder `circleci2/`, make this change there, and then invoke that command in the `kinesis-to-firehose` repo.

**Comment by nathanleiby** [comment]
When circleci2 builds run they connect to a remote VM running Docker. It looks like that Docker VM runs a later version of Docker 

https://discuss.circleci.com/t/update-docker-version/11334

which seems to include the change to deprecate the `--email` flag

https://github.com/moby/moby/pull/20565

**Comment by xavi-** [comment]
Yep, yep this is related to the CircleCI 2 upgrade.  Wasn't sure how to downgrade so I made this change.

Also this change is not reverse compatible since it's removing one of the params.  We could probably rollout this change when ever we decide upgrade all repos to CircleCI 2.

This PR is more of an FYI/what's-the-right-way-to-fix-this request.

File: circleci/docker-publish [hunk 1]
```diff
@@ -55,13 +53,13 @@ fi
 
 # dockerhub login. 
 # `FROM image` in Dockerfile is going to pull from DockerHub
-docker login -u $DOCKER_USER -p $DOCKER_PASS --email="$DOCKER_EMAIL"
+docker login -u $DOCKER_USER -p $DOCKER_PASS
```
Comment (nathanleiby) [inline circleci/docker-publish]: How about just changing this one line?

What about:

```
docker login -u $DOCKER_USER -p $DOCKER_PASS --email="$DOCKER_EMAIL" || docker login -u $DOCKER_USER -p $DOCKER_PASS
```

should be reverse compatible

Comment (nathanleiby) [inline circleci/docker-publish]: @xavi- gave a 👍 for this via Slack 

changed here: 7988db6

**Comment by nathanleiby** [comment]
Verified that it worked in a CircleCI 2 build here: https://circleci.com/gh/Clever/kinesis-to-firehose/150

**Pull request merged**